src/shared/modules/helpers.ts

```diff
@@ -361,3 +362,64 @@ export function hasActionParams(stateAndParams: ItemStateAndParams) {
 
     return hasActionParam(stateAndParams.params);
 }
+
+/**
+ * Collect array of actionParams keys with prefix (_ap_) that must be cleared from widget defaults
+ * @param widgetDefaults
+ * @param queueDataItems
+ * @param itemParams
+ */
+export function getDefaultsActionParamsToClear({
+    widgetDefaults,
+    queueDataItems,
+    itemParams,
+}: {
+    widgetDefaults: StringParams;
+    queueDataItems: StringParams;
+    itemParams: StringParams;
+}) {
+    const widgetDefaultsWithActionParams = hasActionParam(widgetDefaults)
+        ? pickActionParamsFromParams(widgetDefaults)
+        : null;
+
+    if (!widgetDefaultsWithActionParams) {
+        return [];
+    }
+
+    // clear default actionParams if some param with the same name without prefix affected on current widget
+    // ex.: queueDataItems contain {Year: 1970} and default actionParams contain {_ap_Year: 2000}
+    const actionParamKeyToClear: string[] = intersection(
+        Object.keys(queueDataItems),
+        Object.keys(widgetDefaultsWithActionParams),
+    );
+
+    // if itemParams contain {Year: 1970} and actionParam has default widget value {_ap_Year: 2000}
+    // than clear such actionParam
+    const keysForCheckDefaultAvailableValues = intersection(
```

Comment:
I don't understand this logic. Can it be reduced? 
actionParamKeyToClear and keysForCheckDefaultAvailableValues to:

```ts
    const actionParamKeyToClear: string[] = intersection(
        Object.keys({...itemParams, ...queueDataItems}),
        Object.keys(widgetDefaultsWithActionParams),
    );
```

Reply:
discussed case personally, agreed to remove extra check for variant when defaults have not valid params